### ISSUE TITLE
[feature request] hysteresis option in Profile Alarm Rules

### ISSUE THREAD

**Issue by t0mZ4** [thingsboard/thingsboard#7943]
Hi

hysteresis can be a very powerful addition to the Alarm Rules.  It will help to decrease the amount of false-positives alarms produced to our customers.

Programming this into the rule chain will be challenging therefore it can be good if the Profile Alarm Rule has this feature.

feature description:

When alarm is set to trigger at certain threshold (with or without duration), value hysteresis can be defined with an integer value. Only if the Current value crosses the hysteresis value, alarm will trigger.  

Same for clear alarm.

Example:
Set alarm of temperature > 10 degrees, 15 min duration, hysteresis == 1.

Temperature crosses the 10 degrees threshold,  after 15 min alarm will trigger, only if value is >10 + 1.

This is useful to prevent an alarm when a device, such as a freezer, crosses the threshold, but only with 0.xxx of a degree, and after 15-20 minutes it will decrease in temperature.   So at 10.5, no alarm will be produced, but at 11 degrees it will.

finally, hysteresis can be either in integer or %

[an example article from a different system
](**[url](https://service.rotronic.com/manual/understanding-hysteresis.html)**) https://service.rotronic.com/manual/understanding-hysteresis.html

hope this explains it correctly. 

@ashvayka

**Comment by Backdraft007** [comment]
In the device profile you can configure alarm rules and there are options for hysteresis.
![grafik](https://user-images.githubusercontent.com/8741669/212687111-4f60efab-962c-4d19-8470-c5176ad7ff39.png)

Or miss I something?

**Comment by t0mZ4** [comment]
Hi Chris

there is hysteresis for duration (time) but not value or value + time

here is a good article about it
https://www.hw-group.com/support/poseidon-hysteresis-preventing-false-alarms

it shows example for both Hysteresis / time range

thanks!

<https://iot-experts.net/>

Tom Zamir

Owner

***@***.***

+972-55-660-1239 <+972556601239>

Pardes Hanna, Israel

iot-experts.net

<https://www.linkedin.com/r/tomzamir>

**Comment by Backdraft007** [comment]
Yes, in your article it is detailed. But I think for most cases the alarm rule is ok.

**Comment by gcorthey** [comment]
> In the device profile you can configure alarm rules and there are options for hysteresis. ![grafik](https://user-images.githubusercontent.com/8741669/212687445-87d326fe-68cb-4212-bb3c-2b6be75031a2.png)
> 
> Or miss I something?

How can this be done for an asset? I propagate the telemetry to assets and need to manage the alarm for each asset (not device) I don't see how to do that from the rule chain.

**Comment by mykhailokornieiev** [comment]
Hi @t0mZ4 
We already have a feature request in the backlog for the Notifications Histerisis period
Is it a desired feature, or do you need a separate hysteresis period for alarms?

**Comment by mykhailokornieiev** [comment]
I closed this ticket due to inactivity. If this question is still relevant to you, feel free to open a new ticket with a full description of your request.
